site/source/docs/compiling/Building-Projects.rst

```diff
@@ -74,6 +74,41 @@ WebAssembly.
   the configure or cmake scripts.
 
 
+Using CMake
+===========
+
+To use cmake with Emscripten, ``emcmake`` wrapper can be used. A simple ``CMakeLists.txt`` would look something like this,
+
+.. code-block:: cmake
+
+  cmake_minimum_required(VERSION 3.10)
+
+  project(<project_name>)
+
+  add_executable(<target> <sources>)
+  target_link_options(
+      <target>
+      -sDISABLE_EXCEPTION_CATCHING=0
+      -sALLOW_MEMORY_GROWTH
+      -sMODULARIZE
+  )
+
+The Emscripten specific options are added to the ``target_link_options`` command. And assumming we are in a build directory just below the root, the build files can be generated with,
+
+.. code-block:: bash
+
+  emcmake cmake .. -GNinja
+
+The ``emcmake`` wrapper adds the flag, ``-DCMAKE_TOOLCHAIN_FILE=<emscripten cmake toolchain>`` if it is not already present. Subsequently building is just a normal call to ``ninja``, no wrappers required (same for other build systems like ``make``). This would produce the required files in the build directory. CMake automatically adds ``-g`` for Debug builds.
```

Comment:
I don't think `CMake automatically adds -g for Debug builds` is needed is it?    That seems like something that is true in general for cmake and unrelated to emcmake or emscripten.

Reply:
This will produce...

Also maybe move this sentence to before the previous one starting in "Subsequently.."